src/applications/financial-status-report/components/otherExpenses/AddOtherExpense.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  VaTextInput,
+  VaNumberInput,
+} from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import { isValidCurrency } from '../../utils/validations';
+import { MAX_OTHER_LIVING_NAME_LENGTH } from '../../constants/checkboxSelections';
+
+const AddOtherExpense = ({ data, goToPath, setFormData }) => {
+  const { otherExpenses = [] } = data;
+
+  const RETURN_PATH = '/other-expenses-summary';
+
+  // Borrowed from 995 AddIssue
+  // get index from url '/add-issue?index={index}'
+  const searchIndex = new URLSearchParams(window.location.search);
+  let index = parseInt(searchIndex.get('index'), 10);
+  if (Number.isNaN(index)) {
+    index = otherExpenses.length;
+  }
```

Comment:
I was exploring some ideas since we tend to reuse this approach and I was thinking that as a future to reading the URL search params we could do something like:

// OtherExpensesSummary.js

const OtherExpensesSummary = ({
  data,
  goBack,
  goToPath,
  setFormData,
  contentBeforeButtons,
  contentAfterButtons,
}) => {
  // ...

  const onAddExpense = () => {
    const newIndex = otherExpenses.length;
    setFormData({ ...data, expenseIndex: newIndex });
    goToPath('/add-other-expense');
  };

  // ...

  return (
    // ...
    <Link
      className="vads-c-action-link--green"
      onClick={onAddExpense}
    >
      Add additional living expenses
    </Link>
    // ...
  );
};
```

```
// AddOtherExpense.js

const AddOtherExpense = ({ data, goToPath, setFormData }) => {
  const { otherExpenses = [], expenseIndex: index = 0 } = data;

  // ...

  // After successfully adding or updating the expense, clear the expenseIndex from formData
  const newFormData = {
    ...data,
    otherExpenses: newExpenses,
    expenseIndex: undefined, // Clear expenseIndex after use
  };
  setFormData(newFormData);
  goToPath(RETURN_PATH);

  // ...
};

Reply:
I really like this idea! We can definitely experiment with this in the future!